enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -387,23 +542,69 @@ None
 
 ### Implementation Details/Notes/Constraints
 
+#### New API changes for GA release
+
+Below new fields are added in `istiocsrs.operator.openshift.io` to support functionality not included in TP and also
+based on user feedback.
+- `spec.istioCSRConfig.istioDataPlaneNamespaceSelector` - `istio-csr` agent by default creates a configmap with the
+  CA certificate used for signing the istio server certificate and is required by the proxies in the data plane to verify
+  the server certificate to enable mTLS. `istio-csr` agent just creates and updates the configmap `istio-ca-root-cert`
+  and does not delete/removes it. Creating the configmap in all namespaces would not be required for and namespaces created
+  for OpenShift control plane like `kube-system`, and those starting `openshift` can be excluded. `istioDataPlaneNamespaceSelector`
+  helps in filtering the namespaces. Please refer `Risks and Mitigations` section for other details.
+- `spec.istioCSRConfig.certManager.istioCACertificate` - In the TP release, operator was extracting the CA certificate
+  from the configured self-signed issuer and was making it available in the container using volumes. But for issuer types
+  like vault, venafi, the CA certificate used is not available in issuer. `istioCACertificate` allows configuring the configmap
+  which contains the CA certificate used for signing the istiod server certificates, and it will be made available to `istio-csr`
+  using volumes, after basic validations like certificate is indeed CA and the CA attribute in certificates `Basic Constraints`
+  section it set to true and was used for signing the istiod server certificate. When `istioCACertificate` is not provided
+  operator instead of looking in the configured `Issuer/ClusterIssuer` will instead look in the secret containing the
+  certificates obtained from cert-manager for istiod server. The `Certificate` object having the specifics of the istiod
+  server certificate is managed by operator. Please refer `Risks and Mitigations` section for other details.
+  Use case reference: https://issues.redhat.com/browse/CM-564
+- `spec.istioCSRConfig.server.clusterID` - The `clusterID` field is used to configure the Istiod cluster ID. This ensures
+  that only certificate signing requests (CSRs) from the matching Istio control plane are allowed.
+
+Validations for fields in the `istiocsrs.operator.openshift.io` API are updated with minimum and maximum limits to
+restrict the values that can be configured, preventing users from setting arbitrarily large data that could cause
+system instability.
+For below fields, the lower and upper bounds have been set based on what could be considered as reasonable, as there is
+no defined standard. And can be updated in future based on user feedback.
```

Comment:
We need to be careful about the validations here. We should not choose random values; it should be backed by valid reasoning as to why we chose these numbers. 
I remember some past discussion about selecting the right `MaxLength`; otherwise, it would be a breaking change for the users once they will upgrade the operator.

Reply:
I understand the concern, and lowering the limit would cause backward compatibility issues. But do you foresee any issue with the limitations set on the particular fields proposed here?